common/src/main/java/com/wynntils/models/gear/SetModel.java

```diff
@@ -0,0 +1,154 @@
+    public static final Pattern SET_PATTERN = Pattern.compile("§a(.+) Set §7\\((\\d)/\\d\\)");
+
+    // Stored as a map for quick lookup <name, SetInfo>
+    private final Map<String, SetInfo> setData = new HashMap<>();
+    private final Map<GearSlot, SetInstance> setInstances = new EnumMap<>(GearSlot.class);
+
+    public SetModel() {
+        super(List.of());
+        loadSetData();
+    }
+
+    public SetInfo getSetInfo(String setId) {
+        return setData.getOrDefault(setId, null);
+    }
+
+    public SetInfo getSetInfoForItem(String itemName) {
+        return getSetInfo(getSetName(itemName));
+    }
+
+    /**
+     * @param itemName The name of the item to check
+     * @return The set name if the item is part of a set, null otherwise
+     */
+    public String getSetName(String itemName) {
+        for (Map.Entry<String, SetInfo> entry : setData.entrySet()) {
+            if (entry.getValue().items().contains(itemName)) {
+                return entry.getKey();
+            }
+        }
+        return null;
+    }
+
+    public void updateSetInstance(GearSlot slot, SetInstance instance) {
+        setInstances.put(slot, instance);
+    }
+
+    /**
+     * @return A Set of all equipped set names
+     */
+    public Set<String> getUniqueSetNames() {
+        return setInstances.values().stream().map(x -> x.getSetInfo().name()).collect(Collectors.toSet());
+    }
+
+    public int getTrueCount(String setName) {
+        int trueCount = 0;
+
+        for (ItemStack itemStack : McUtils.inventory().armor) {
+            for (StyledText line : LoreUtils.getLore(itemStack)) {
+                Matcher nameMatcher = SetModel.SET_PATTERN.matcher(line.getString());
+                if (nameMatcher.matches() && nameMatcher.group(1).equals(setName)) {
+                    trueCount++;
+                }
+            }
+        }
+
+        int[] accessorySlots = {9, 10, 11, 12};
+        if (McUtils.player().hasContainerOpen()) {
+            // Scale according to server chest size
+            // Eg. 3 row chest size = 27 (ends on i=26 since 0-index), we would get accessory slots {27, 28, 29, 30}
+            int baseSize = McUtils.player().containerMenu.getItems().size();
+            accessorySlots = new int[] {baseSize, baseSize + 1, baseSize + 2, baseSize + 3};
+        }
+        for (int i : accessorySlots) {
+            for (StyledText line : LoreUtils.getLore(McUtils.inventory().getItem(i))) {
+                Matcher nameMatcher = SetModel.SET_PATTERN.matcher(line.getString());
+                if (nameMatcher.matches() && nameMatcher.group(1).equals(setName)) {
+                    trueCount++;
+                }
+            }
+        }
+
+        Optional<WynnItem> wynnItem =
+                Models.Item.getWynnItem(McUtils.player().getItemInHand(InteractionHand.MAIN_HAND));
+        if (wynnItem.isPresent() && wynnItem.get() instanceof GearItem gearItem && gearItem.meetsActualRequirements()) {
```

Comment:
This is basically duplicated code. Perhaps add a method in `PlayerInventoryModel`, `getWornItems` or something along the lines?